plugins/processors/flux/flux.go

```diff
@@ -0,0 +1,342 @@
+package flux
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/csv"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/lang"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/telegraf/plugins/processors/flux/tables"
+	// Cannot import builtin because we need to register telegraf.from().
+	_ "github.com/influxdata/flux/stdlib"
+	"github.com/influxdata/flux/values"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+const (
+	description  = "Runs a flux script to process inputs and produce outputs"
+	sampleConfig = `
+  ## Path to a Flux script.
+  path = "path/to/script.flux"
+  ## Enables debug mode if an output file is specified.
+  debug = "path/to/out.csv"
+`
+	measurementColumnLabel = "_measurement"
+)
+
+type Flux struct {
+	Path  string `toml:"path"`
+	Debug string `toml:"debug"`
+
+	Log telegraf.Logger `toml:"-"`
+
+	debugFile *os.File
+	program   flux.Program
+}
+
+func init() {
+	processors.Add("flux", func() telegraf.Processor {
+		return &Flux{}
+	})
+}
+
+func (p *Flux) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *Flux) Description() string {
+	return description
+}
+
+func (p *Flux) init() error {
+	if err := p.compile(); err != nil {
+		return fmt.Errorf("script failed to compile: %v", err)
+	}
+	if len(p.Debug) > 0 && p.debugFile == nil {
+		f, err := os.OpenFile(p.Debug, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+		if err != nil {
+			return fmt.Errorf("could not open debug file: %s", p.Debug)
+		}
+		p.debugFile = f
+	}
+	return nil
+}
+
+func (p *Flux) Apply(metrics ...telegraf.Metric) []telegraf.Metric {
+	if len(metrics) == 0 {
+		return nil
+	}
+	if err := p.init(); err != nil {
+		p.Log.Errorf("Error on flux processor init: %v", err)
+		return nil
+	}
+
+	tbls, err := getTables(metrics...)
+	if err != nil {
+		p.Log.Errorf("Could not extract tables from metrics: %v", err)
+		return nil
+	}
+	if len(p.Debug) > 0 {
```

Comment:
Let's play save and change this to
```suggestion
	if p.debugFile != nil {
```